### PR TITLE
script: static address taproot script

### PR DESCRIPTION
**Pull request by hieblmi**
In this PR we introduce taproot scripting for static loop-in addresses.

### CODE REVIEW

File: staticaddr/script/script_test.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+					sigHashes, value, prevOutFetcher,
+				)
+			}
+
+			assertEngineExecution(t, testCase.valid, newEngine)
+		})
+	}
+}
+
+func MuSig2Sign(version input.MuSig2Version, privKeys []*btcec.PrivateKey,
```
Comment (bhandras) [inline staticaddr/script/script_test.go:198]: I wonder if it's worth moving this to say `utils` package?

**Review by bhandras — APPROVED** [review]
LGTM 🎉

File: staticaddr/script/script_test.go (hunk 1 above)
Comment (hieblmi) [reply staticaddr/script/script_test.go:198]: There's a `utils.go` and a `utils/htlc_utils.go`. Should we create `utils/musig.go`. Although sometimes utils is an anti pattern.

Comment (bhandras) [reply staticaddr/script/script_test.go:198]: Yeah sgtm, it is somewhat of a too generic name I agree, but at least it is reusable.

Comment (hieblmi) [reply staticaddr/script/script_test.go:198]: Ideally the server would also use this functionality.

File: staticaddr/script/script.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+const (
+	// TaprootMultiSigWitnessSize evaluates to 66 bytes:
+	//	- num_witness_elements: 1 byte
+	//	- sig_varint_len: 1 byte
+	//	- <sig>: 64 bytes
+	TaprootMultiSigWitnessSize = 1 + 1 + 64
+
+	// TaprootExpiryScriptSize evaluates to 39 bytes:
+	//	- OP_DATA: 1 byte (trader_key length)
+	//	- <trader_key>: 32 bytes
```
[the remark this answers is not in the document]
Comment (hieblmi) [reply staticaddr/script/script.go:23]: ...from pool? Will change it _caugh_ _caugh_.

File: staticaddr/script/script.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+	// TaprootKey is the taproot public key which is created with the above
+	// 3 inputs.
+	TaprootKey *btcec.PublicKey
+
+	// RootHash is the root hash of the taptree.
+	RootHash chainhash.Hash
+}
+
+// NewStaticAddress constructs a static address script.
+func NewStaticAddress(muSig2Version input.MuSig2Version, csvExpiry int64,
```
[the remark this answers is not in the document]
Comment (hieblmi) [reply staticaddr/script/script.go:63]: done

**Pull request merged**